catwalk/models/huggingface.py

```diff
@@ -98,11 +105,131 @@ def _predict_mc(
                 )
                 results = model(
                     return_dict=True,
-                    **{key: tensor.view(len(batch), number_of_choices, -1).to(device) for key, tensor in tensors.items()})
+                    **{
+                        key: tensor.view(len(batch), number_of_choices, -1).to(model.device)
+                        for key, tensor in tensors.items()
+                    })
                 for instance, logits in zip(batch, results.logits.detach().cpu()):
                     yield {
                         "correct_answer_index": instance.correct_answer_index,
                         "logits": logits,
                         "acc": (logits, instance.correct_answer_index),
                     }
-                    
\ No newline at end of file
+
+    def trainable_copy(self) -> "TrainableHFAutoModel":
+        return TrainableHFAutoModel(self.pretrained_model_name_or_path)
+
+
+class TrainableHFAutoModel(TrainableModel):
+    VERSION = "003stp"
+
+    def __init__(self, pretrained_model_name_or_path: str):
+        super().__init__(None)
+        self.tokenizer = cached_transformers.get_tokenizer(AutoTokenizer, pretrained_model_name_or_path)
+
+        # This is a bit messy because huggingface doesn't support multi-task training.
+        # We instantiate both MC and QA models, and then we set the inner modules (usually the transformer itself)
+        # to be the same.
+        self.mc_model = cached_transformers.get(AutoModelForMultipleChoice, pretrained_model_name_or_path, True)
```

Comment:
Are we only supporting QA and MC models? If so, we should rename the classes to make that clear.

Reply:
At the moment we do. NLI is already in the works, too. The plan is to fill out these classes to support all the tasks.